data/core/macros/items.cfg

```diff
@@ -261,11 +261,22 @@
 #enddef
 
 #define OBJ_POTION_HOLY X Y ID
-    {PICKUPPABLE_ITEM {ID} {X} {Y} side=1 items/holy-water.png
+    {PICKUPPABLE_ITEM {ID} {X} {Y} (
+        side=1
+        [has_attack]
+            range=melee
+        [/has_attack]
+        [not]
+            [has_attack]
+                range=melee
+                type=arcane
+            [/has_attack]
+        [/not]
+    ) items/holy-water.png
```

Comment:
You should be able to combine both [has_attack] tags here, since [not] is also supported inside attack filters.